samcli/lib/build/app_builder.py

```diff
@@ -611,3 +629,73 @@ def _parse_builder_response(stdout_data: str, image_name: str) -> Dict:
             raise ValueError(msg)
 
         return cast(Dict, response)
+
+    @staticmethod
+    def _get_env_vars_value(filename: Optional[str]) -> Any:
+        """
+        If the user provided a file containing values of environment variables, this method will read the file and
+        return its value
+
+        :param string filename: Path to file containing environment variable values
+        :return dict: Value of environment variables, if provided. None otherwise
+        :raises IOError: If the file was not found or not a valid JSON
+        """
+        if not filename:
+            return None
+
+        # Try to read the file and parse it as JSON
+        try:
+
+            with open(filename, "r") as fp:
```

Comment:
This will break on Windows with UTF-8 bytes. See https://github.com/aws/aws-sam-cli/pull/2583, #2312, the linked issues, and likely many other unreported ones.

Reply:
Got it, will update this, thanks!